plugins/cpu/fu-plugin-cpu.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2020 Mario Limonciello <mario.limonciello@dell.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+#include "fu-cpu-device.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+gboolean
+fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+{
+	gsize length;
+	g_autofree gchar *data = NULL;
+	g_auto(GStrv) lines = NULL;
+
+	if (!g_file_get_contents ("/proc/cpuinfo", &data, &length, error))
+		return FALSE;
+
+	lines = g_strsplit (data, "\n\n", 0);
+	for (guint i = 0; lines[i] != NULL; i++) {
+		g_autoptr(FuCpuDevice) dev = NULL;
+		g_strstrip(lines[i]);
+		if (strlen (lines[i]) == 0)
+			continue;
+		dev = fu_cpu_device_new (lines[i]);
```

Comment:
I still think you need to keep track of the last physical ID -- otherwise I'm going to get 64 devices deduped by the daemon, no?

Reply:
Isn't that the point of daemon de-duping?  If you think it should be in plugin though, can do it there too again.

Reply:
True. Adding 63 extra devices to the daemon isnt free, but I wonder if it's just premature optimization on my part. How noisy is is?

Reply:
With debug logs turned on we're looking at about 5 extra debugging lines per CPU and I guess between 0 and 1/1000 of a second time per CPU?  I don't know the unit after seconds in the debug logs.

Reply:
Fixed up the other two now.  I guess try the branch on your side and see if you think it's too much.